### ISSUE TITLE
Parsing bug for `foo<typeof a[0].b>()`

### ISSUE THREAD

**Issue by devanshj** [microsoft/TypeScript#44857]
# Bug Report

### 🔎 Search Terms

Parsing, typeof, generic, index access

### 🕗 Version & Regression Information

Tested with 4.3.4

### ⏯ Playground Link

[Playground](https://www.typescriptlang.org/play?jsx=0#code/DYUwLgBAhhC8EG0DeEBGAuCBGCBfAutAM4QDGA9gHZFgBQokAZueXBADwAqEIAHmCEoATElgB8ACgCUcMRABE82rWbl2YAJ4AHEOUbQEABnwA6VJKkBuWgHobEAKqU+O0gKE8ATp-Kf0t+wB5HU8oMF8IAHJ2SLIoSkpySFQQaC0tYABLEA9wiE0dEmjuPgFhUQtZCBpPTMoAc1j4j0iausaTCAkAJgBmADYAVillBggALzYoI1NUFRZ1bV19cYsgA)

### 💻 Code

```ts
let a = [{ b: 1 }] as const
let foo = <T extends 1>() => ""

foo<typeof a[0].b>();
// Unexpected error:
// Operator '<' cannot be applied to types '<T extends 1>() => string' and 'string'. (2365)

let z = a[0].b
foo<typeof z>()
```

### 🙁 Actual behavior

There is an error at line 4 - `Operator '<' cannot be applied to types '<T extends 1>() => string' and 'string'. (2365)`

### 🙂 Expected behavior

There should be no errors.

**Comment by whzx5byb** [comment]
In your example, the `typeof` operator appears in a type context (not expression context!), and you should use indexed access type `foo<typeof a[0]['b']>()` to get the type of a property.

**Comment by devanshj** [comment]
Umm I don't think so, if it were in a type context `typeof a.b.c` would also not work but [it does](https://www.typescriptlang.org/play?jsx=0#code/DYUwLgBAhhC8EG8ICMBciIGN0EYIF8DoBnLAewDtiwAoUSAMzLLggB4AVCEADzBAoATUjgB8ACgCUcURABEcmjSZk2YAJ4AHEGQbQAdMn2YJkgNxA).

**Comment by DanielRosenwasser** [comment]
Seems possibly related to https://github.com/microsoft/TypeScript/issues/19707.

My understanding is that we try parsing type arguments, we see `typeof a[0].b` which is initially parsed as `(typeof a)[0]`. The `.` can't follow the type though because the left side is not valid for a qualified name though, so we determine that we can't parse the type arguments and throw the work away, back-tracking and trying to parse it as a `<` comparison.

**Comment by sandersn** [comment]
I agree, this is a duplicate of #19707
